Content.Server/GhostTypes/MindRememberBodySystem.cs

```diff
@@ -0,0 +1,27 @@
+using Content.Shared.Damage;
+using Content.Shared.Mind;
+using Content.Shared.Mind.Components;
+
+namespace Content.Server.GhostTypes;
+
+public sealed class MindRememberBodySystem : EntitySystem
+{
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<MindRememberBodyComponent, BeforeBodyDestructionEvent>(SaveBody);
+    }
+
+    /// <summary>
+    /// Saves the damage of a player's previous body inside their MindComponent
+    /// </summary>
+    private void SaveBody(EntityUid uid, MindRememberBodyComponent component, BeforeBodyDestructionEvent args)
```

Comment:
This should be in the `Entity<T>` style mentioned earlier. It'd end up being `private void SaveBody(Entity<MindRememberBodyComponent> ent, ref BeforeBodyDestructionEvent args)`